tests/integration/mount_path_test.go

```diff
@@ -0,0 +1,139 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/kube"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/smbclient"
+)
+
+type MountPathSuite struct {
+	suite.Suite
+
+	auths                   []smbclient.Auth
+	commonSources           []kube.FileSource
+	smbshareSetupSources    []kube.FileSource
+	smbshareSources         []kube.FileSource
+	smbShareSetupResource   types.NamespacedName
+	setupServerLabelPattern string
+	smbShareResource        types.NamespacedName
+	serverLabelPattern      string
+	tc                      *kube.TestClient
+}
+
+func (s *MountPathSuite) getPodIP(labelPattern string) (string, error) {
```

Comment:
In the future we probably want to decrease the number of tests the rely primarily on reaching the smb share via the pods IP address. It's not an issue for now, but something to keep in mind for later.

Reply:
I simply used the ip address as that showed up on the first example I looked at. I have added a new patch which removes the use of the ip address. I hope you don't mind me keeping this as a separate patch as I can use these as a reference for when I add more tests.

Reply:
I think that'll be fine.